db/seeds/development/competitions.seeds.rb

```diff
@@ -29,7 +29,7 @@ def random_city(country)
 
   countries = Country.all
 
-  delegate = User.find_by(delegate_status: "delegate")
+  delegate = RolesMetadataDelegateRegions.includes(:user_role).where.not(status: RolesMetadataDelegateRegions.statuses[:trainee_delegate]).select { |role_metadata| role_metadata.user_role.is_active? }.sample.user
```

Comment:
Using Rails `enum` gives you `not_*` scopes for free, so using `RolesMetadataDelegateRegions.not_trainee_delegate` should work.

Reply:
Also, you can use the `active` scope on `UserRole` to execute this query directly on DB level. The magic feature is called `merge`, which can combine arbitrary `scope` definitions:
```suggestion
  delegate = RolesMetadataDelegateRegions.joins(:user_role).not_trainee_delegate.merge(UserRole.active).sample.user
```

Reply:
Wow thanks a lot, this is nice. Accepted the suggestion.